src/game/GameHud.cpp

```diff
@@ -0,0 +1,446 @@
+ * Consumable slot is only visible when the player has a consumable item in their inventory.
+ */
+void GameHud::renderConsumable(const SDL_Rect& screenRect, const Player& p) {
+
+    if (screenRect.w <= screenRect.h) {
+        consumableSlot.w =  screenRect.w * CONSUMABLE_SIZE_RAT;
+        consumableSlot.h =  screenRect.w * CONSUMABLE_SIZE_RAT;
+    } else {
+        consumableSlot.w = screenRect.h * CONSUMABLE_SIZE_RAT;
+        consumableSlot.h = screenRect.h * CONSUMABLE_SIZE_RAT;
+    }
+
+    consumableSlot.x = screenRect.w * PADDING_RAT;
+    consumableSlot.y = screenRect.h - screenRect.w * PADDING_RAT - consumableSlot.h;
+
+    Renderer::instance().render(consumableSlot, TEXTURES::CONSUMABLE_SLOT);
+    Renderer::instance().render(consumableSlot, TEXTURES::HEALTHPACK);
+
+}
+
+/**
+ * Function: renderWeaponSlots
+ *
+ * Date:
+ * JF: April 1, 2017:
+ *
+ * Designer:
+ * Jacob Frank
+ *
+ * Programmer:
+ * Jacob Frank
+ *
+ * Interface: renderWeaponSlots(SDL_Rect screenRect, Player p)
+ *                  SDL_Rect screenRect: The Current screen properties (height, width)
+ *                  Player p: The player in the game. used to find current equipped weapon
+ *
+ * Notes:
+ * Function, when called renders the Weapon inventory slots along the bottom center of the visible screen
+ * Inventory slots are only visible when the player changes their currently equiped weapon.
+ */
+void GameHud::renderWeaponSlots(const SDL_Rect& screenRect, const Player& p) {
+    const int weaponSlotWidth = screenRect.w * WEAPON_SLOT_WIDTH_RAT;
+    const int weaponSlotHeight = screenRect.h * WEAPON_SLOT_HEIGHT_RAT;
+    const int weaponSlotPosY = screenRect.h - screenRect.w * PADDING_RAT - weaponSlotHeight;
+
+
+    inventorySlot[1].w = weaponSlotWidth;
+    inventorySlot[1].h = weaponSlotHeight;
+    inventorySlot[1].y = weaponSlotPosY;
+    inventorySlot[1].x = screenRect.w * HALF - inventorySlot[0].w * HALF; //Positions slot 1 in the center
+
+    //positions slot 0 to the left of slot 1
+    inventorySlot[0].w = weaponSlotWidth;
+    inventorySlot[0].h = weaponSlotHeight;
+    inventorySlot[0].y = weaponSlotPosY;
+    inventorySlot[0].x = inventorySlot[1].x - weaponSlotWidth - screenRect.w * PADDING_RAT;
+
+    //positions slot 2 to the right of slot 1
+    inventorySlot[2].w = weaponSlotWidth;
+    inventorySlot[2].h = weaponSlotHeight;
+    inventorySlot[2].y = weaponSlotPosY;
+    inventorySlot[2].x = inventorySlot[1].x + weaponSlotWidth + screenRect.w * PADDING_RAT;
+
+    //Decrease the opacity of the inventory item slots by 1
+    //This gets called on every loop resulting in the inventory items fading away
+    decrementOpacity(1);
+
+    SDL_SetTextureAlphaMod(Renderer::instance().getTexture(static_cast<int>(TEXTURES::ACTIVE_SLOT)),
+        inventorySlotOpacity);
+    SDL_SetTextureAlphaMod(Renderer::instance().getTexture(static_cast<int>(TEXTURES::PASSIVE_SLOT)),
+        inventorySlotOpacity);
```

Comment:
not a blocker but do note that there is now a set alpha wrapper in renderer directly as an fyi